### PR TITLE
Issue #1696: Use Reply-To headers to prevent contact module mail from…

### PR DESCRIPTION
**Pull request by jenlampton**
Fixes https://github.com/backdrop/backdrop-issues/issues/1696

### CODE REVIEW

File: core/includes/mail.inc [hunk 1]
```diff
@@ -155,6 +155,34 @@ function backdrop_mail($module, $key, $to, $language, $params = array(), $reply
   }
   $message['headers'] = $headers;
 
+  // If requested set Reply-To to the From and re-format From address.
+  if (variable_get('site_mail_replyto') &&
+      empty($message['headers']['Reply-To']) &&
+      ($default_from_name = variable_get('site_name', 'Backdrop CMS')) &&
```
Comment (quicksketch) [inline core/includes/mail.inc:161]: `variable_get()` needs updating.

File: core/includes/mail.inc [hunk 2]
```diff
@@ -155,6 +155,34 @@ function backdrop_mail($module, $key, $to, $language, $params = array(), $reply
...
+      ($default_from_parts = explode('@', $default_from)) &&
+      count($default_from_parts) == 2 &&
+      $default_from_parts[1] &&
+      stripos($message['from'], '@' . $default_from_parts[1]) === FALSE) {
```
Comment (quicksketch) [inline core/includes/mail.inc:165]: This is one of the ugliest IF statements I've seen in a while. Let's add some nesting if needed and avoid the inline variable assignments.

Comment (jenlampton) [inline core/includes/mail.inc:165]: I didn't write it, just applied a patch :D

**Comment by jenlampton** [comment]
Replaced by https://github.com/backdrop/backdrop/pull/1823

**Pull request closed without merging**